core/tests/protractor_utils/AdminPage.js

```diff
@@ -292,6 +334,190 @@ var AdminPage = function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Similarity upload failed – download similarities button not clickable.');
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Download similarity file button not clickable');
+    await similarityDownloadButton.click();
+  };
+
+  this.clearSearchIndex = async function() {
+    await waitFor.elementToBeClickable(searchIndexClearButton,
+      'Clear search index button not clickable');
+    await searchIndexClearButton.click();
+    await general.acceptAlert();
+  };
+
+  this.expectSearchIndexToBeCleared = async function() {
+    await waitFor.visibilityOf(statusMessage,
+      'Status message not showing up.');
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+    expect(statusMessage.getText()).toEqual(
+      'Index successfully cleared.');
+  };
+
+  this.flushMigrationBotContributions = async function() {
+    await waitFor.elementToBeClickable(flushMigrationBotContributionsButton,
+      'Migration bot flush contributions button not clickable');
+    await flushMigrationBotContributionsButton.click();
+    await general.acceptAlert();
+  };
+
+  this.expectMigrationBotContributionsToBeFlushed = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Migration bot contributions successfully flushed.',
+      'Migration bot cuntributions not flushing.');
+    return true;
+  };
+
+  this.regenerateContributionsForTopic = async function(topicId) {
+    await waitFor.visibilityOf(regenerateContributionsTopicIdInput,
+      'Regenerate contributions topic ID input not showing up.');
+    await regenerateContributionsTopicIdInput.sendKeys(topicId);
+    await waitFor.elementToBeClickable(regenerateContributionsSubmitButton,
+      'Regenerate conributions form submit button not clickable');
+    await regenerateContributionsSubmitButton.click();
+    await general.acceptAlert();
+    await waitFor.visibilityOf(regenerateContributionsTopicIdInput,
+      'Regenerate contributions topic ID input not showing up.');
+    await regenerateContributionsTopicIdInput.clear();
+  };
+
+  this.expectRegenerationOutcome = async function(isMeantToSucceed,
+      topicId = '0') {
+    if (!isMeantToSucceed) {
+      var text = 'Server error: Entity for class TopicModel with id ' +
+        topicId + ' not found';
+    } else {
+      var text = 'No. of opportunities model created: 0';
+    }
+    await waitFor.visibilityOf(regenerationMessage,
+      'Regeneration message not visible');
+    expect(regenerationMessage.getText()).toEqual(text);
+  };
+
+  this.expectConributionsToBeRegeneratedForTopic = async function() {
+    var text = 'No. of opportunities model created: 0';
+    await waitFor.visibilityOf(regenerationMessage,
+      'Regeneration message not visible');
+    expect(regenerationMessage.getText()).toEqual(text);
```

Comment:
This seems redundant with the end of `this.expectRegenerationOutcome`

Reply:
I've deleted it